bitnami/keycloak/templates/statefulset.yaml

```
@@ -59,9 +59,45 @@ spec:
       {{- if .Values.podSecurityContext.enabled }}
       securityContext: {{- omit .Values.podSecurityContext "enabled" | toYaml | nindent 8 }}
       {{- end }}
-      {{- if .Values.initContainers }}
+      {{- if or .Values.initContainers .Values.auth.tls.enabled }}
       initContainers: {{- include "common.tplvalues.render" (dict "value" .Values.initContainers "context" $) | nindent 8 }}
       {{- end }}
+      {{- if .Values.auth.tls.enabled }}
+      {{- $fullname := include "keycloak.fullname" . }}
+        - name: init-certs
```

There is an `if` missing here:
```suggestion
      initContainers: 
        {{- if .Values.initContainers }}
        {{- include "common.tplvalues.render" (dict "value" .Values.initContainers "context" $) | nindent 8 }}
        {{- end }}
        {{- if .Values.auth.tls.enabled }}
        {{- $fullname := include "keycloak.fullname" . }}
        - name: init-certs
```

---

Line 62 handles the logical condition for that `if` statement. Please double check, but your suggested change should not be required.

---

My bad, I didn't explain myself very well. The `if` I was referring to is `{{- if .Values.initContainers }}`. As it is now, whenever `auth.tls.enabled=true`, the code `{{- include "common.tplvalues.render" (dict "value" .Values.initContainers "context" $) | nindent 8 }}` is going to be executed, which is not something necessary. IMHO, separating this in 2 `ifs` leaves a clearer code and is the default practice in most of our charts.

---

Ah ok I see now. You are correct my logic isn't perfect there. I agree two `if` statements needed to keep the extra include from generating.